pkg/registry/storage/csinode/strategy.go

```diff
@@ -54,15 +62,13 @@ func (csiNodeStrategy) Validate(ctx context.Context, obj runtime.Object) field.E
 	return errs
 }
 
-// Canonicalize normalizes the object after validation.
 func (csiNodeStrategy) Canonicalize(obj runtime.Object) {
 }
 
 func (csiNodeStrategy) AllowCreateOnUpdate() bool {
 	return false
 }
 
-// PrepareForUpdate sets the Status fields which is not allowed to be set by an end user updating a CSINode
 func (csiNodeStrategy) PrepareForUpdate(ctx context.Context, obj, old runtime.Object) {
```

Comment:
Sorry I meant that the logic here for dropping the allocatable field should check for feature gate and if the field is already set.  You can use the [storageclass prepareforupdate](https://github.com/kubernetes/kubernetes/blob/d75da7e17e234277f072f3e1324d8a475343e26a/pkg/registry/storage/storageclass/strategy.go#L66) as an example.

Reply:
Done. I used a map to avoid repeatedly iterating over the drivers of the old object.